test/system/test_misc.py

```diff
@@ -788,3 +788,54 @@ def test_catch_bug_804():
 
     for trap in model.traps:
         assert trap.id is not None
+
+
+def test_soret_surface_flux_mass_balance():
+    """
+    Test to catch bug 830
+
+    1D steady state simulation with Soret on.
+    Compute the surface flux on both surfaces.
+    The surface fluxes should be equal in magnitude
+    but opposite in sign.
+
+    The case is made so that the concentration profile
+    is flat (ie. diffusive flux is zero) but there's a
+    high T gradient (ie. Soret flux is not zero).
+    """
+    my_model = F.Simulation()
+
+    my_model.mesh = F.MeshFromVertices(vertices=np.linspace(0, 0.05, num=100))
```

Comment:
Have you checked cylindrical/spherical? I suppose it should be ok.

Reply:
I haven't but I'll see if I can parametrise this test